src/com/ms/silverking/cloud/dht/config/WeightsTable.java

```diff
@@ -10,65 +10,65 @@
 import java.util.Map;
 
 public class WeightsTable {
-    private final Map<String, Double>    nodeWeights;
-    
-    public WeightsTable(Map<String, Double>    nodeWeights) {
-        this.nodeWeights = nodeWeights;
-    }
-    
-    public static WeightsTable parse(File fileName) throws IOException {
-        return parse(new FileInputStream(fileName));
-    }
-    
-    public static WeightsTable parse(InputStream inStream) throws IOException {
-        try {
-            BufferedReader    reader;
-            String            line;
-            Map<String, Double>    nodeWeights;
-            
-            nodeWeights = new HashMap<String, Double>();
-            reader = new BufferedReader(new InputStreamReader(inStream));
-            do {                
-                line = reader.readLine();
-                if (line != null) {
-                    line = line.trim();
-                    if (line.length() != 0) {
-                        String[]    tokens;
-                        
-                        tokens = line.split("\\s+");
-                        if (tokens.length != 2) {
-                            throw new IOException("Invalid weight line: "+ line);
-                        } else {
-                            nodeWeights.put(tokens[0], Double.parseDouble(tokens[1]));
-                        }
-                    }
-                }
-            } while (line != null);
-            return new WeightsTable(nodeWeights);
-        } finally {
-            inStream.close();
-        }
-    }
-    
-    public void display() {
-        for (Map.Entry<String, Double> entry : nodeWeights.entrySet()) {
-            System.out.println(entry.getKey() +"\t"+ entry.getValue());
-        }
-    }
-    
-    // for unit testing
-    public static void main(String[] args) {
-        try {
-            if (args.length != 1) {
-                System.err.println("<file>");
+  private final Map<String, Double> nodeWeights;
+
+  public WeightsTable(Map<String, Double> nodeWeights) {
+    this.nodeWeights = nodeWeights;
+  }
+
+  public static WeightsTable parse(File fileName) throws IOException {
+    return parse(new FileInputStream(fileName));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when WeightsTable.parse throws an exception. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).